### ISSUE TITLE
Terraform apply fails with message 'registry.terraform.io/hashicorp/local: there is no package for registry.terraform.io/hashicorp/local 2.1.0 cached in .terraform/providers'

### ISSUE THREAD

**Issue by lamontadams** [hashicorp/setup-terraform#139]
Greetings. I have this workflow
```
name: Dev Deployment - Manual

on: workflow_dispatch

jobs:
  deploy:
    runs-on: ubuntu-latest
    name: deploy
    steps:
      - name: Checkout Branch
        uses: actions/checkout@v2

      - name: Setup Node
        uses: actions/setup-node@v2
        with:
          node-version: 14

      - name: Install pnpm
        run: npm install -g pnpm && pnpm config set script-shell "/usr/bin/bash"

      - name: Sanity Check
        run: |
          echo branch: `git branch --show-current`
          echo node: `node -v`
          echo pnpm: `pnpm -v`
          echo git diff: `git --no-pager diff HEAD^`

      - name: Set Git Config
        run: |
          git config pull.rebase false
          git config --global user.email "release-workflow@openbase.io"
          git config --global user.name "Release Workflow"
          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}

      - name: Cache Node Modules
        id: node-cache
        uses: actions/cache@v2
        env:
          cache-name: cache-node-modules
        with:
          path: ./node_modules
          key: ${{ env.cache-name }}-${{ hashFiles('pnpm-lock.yml') }}

      - name: Initliaze .npmrc
        run: echo -e "//npm.pkg.github.com/:_authToken=${{ secrets.OPENBASE_NPM_TOKEN }}\n$(cat .npmrc)" > .npmrc

      - name: pnpm install
        run: pnpm install --frozen-lockfile

      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_DEPLOY_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_DEPLOY_SECRET }}
          aws-region: us-east-1

      - name: HashiCorp - Setup Terraform
        uses: hashicorp/setup-terraform@v1.2.1

      - name: Deploy
        env:
          ALGOLIA_API_KEY: ${{ secrets.ALGOLIA_WRITE_API_KEY }}
          ALGOLIA_APP_ID: ${{ secrets.ALGOLIA_APP_ID }}
          NODE_ENV: dev
          DEPLOY_ENV: dev
        run: pnpm deploy
```

`pnpm deploy` executes this shell script
```
#!/bin/bash
set -euo pipefail
cd ./tf
terraform init -no-color
terraform workspace select $DEPLOY_ENV -no-color || terraform workspace new $DEPLOY_ENV -no-color
terraform apply -input=false -no-color -auto-approve
```

This setup worked wonderfully for several weeks when run as a github action. It continues to work for me when run locally. Today I noticed workflow dispatch on the github action ends with this error output, despite **output from a successful terraform init** appearing in log before. I've googled but haven't found anything relevant, and a search of issues here has also not found any results. Any suggestions are welcome.
 
```
Error: Required plugins are not installed

The installed provider plugins are not consistent with the packages selected
in the dependency lock file:
  - registry.terraform.io/hashicorp/local: there is no package for registry.terraform.io/hashicorp/local 2.1.0 cached in .terraform/providers

Terraform uses external plugins to integrate with a variety of different
infrastructure services. To download the plugins required for this
configuration, run:
  terraform init

Error: Terraform exited with code 1.
/home/runner/work/_temp/6b4166c2-02af-4c65-b68b-226728c10f88/terraform-bin workspace new dev -no-color

Error: Required plugins are not installed

The installed provider plugins are not consistent with the packages selected
in the dependency lock file:
  - registry.terraform.io/hashicorp/local: there is no package for registry.terraform.io/hashicorp/local 2.1.0 cached in .terraform/providers

Terraform uses external plugins to integrate with a variety of different
infrastructure services. To download the plugins required for this
configuration, run:
  terraform init
```

**Comment by ivanRylach** [comment]
First of all, thank you team for driving this awesome project!

At the same time, within the last hour `terraform validate` started failing when using `hashicorp/setup-terraform@v1` GitHub action

```
Error: registry.terraform.io/hashicorp/random: there is no package for registry.terraform.io/hashicorp/random 3.1.0 cached in .terraform/providers
```

**Comment by ivanRylach** [comment]
Resolved my case by cleaning up `.terraform.lock.hcl`

**Comment by samruthr** [comment]
@ivanRylach what did you end up cleaning out the lock file?

**Comment by artfisica** [comment]
Thanks, @ivanRylach and @samruthr 

I had the same problem just now and I was getting a bit crazy. I solved the issue **removing** the `.terraform.lock.hcl`, and let the GitHub Action re-create again.

But this is not obvious at all (unless for me)

**Comment by ab** [comment]
I ran in to this error message when I had an unused provider in the lockfile. This smells like a new Terraform bug.

To reproduce:
- A previous resource used some provider (like hashicorp/null)
- All resources using that provider have been removed
- The provider is still listed in `.terraform.lock.hcl`

To fix:
- Remove the unused provider from `.terraform.lock.hcl`

**Comment by mersedsv** [comment]
I had the same issue as well. Resolved it by removing .lock file and regenerating it with "terraform plan"

**Comment by danvau7** [comment]
Have had a couple of issues with Terraform tutorials where the solution is to delete the `.terraform.lock.hcl` file and run `terraform init` again (which creates a correct lock file with the correct lock values on rerun).

**Comment by szEvEz** [comment]
This is supposed to be fixed with version v1.1.3 - https://github.com/hashicorp/terraform/releases/tag/v1.1.3

**Comment by bflad** [comment]
Hi folks 👋 This sounds like it was resolved upstream in Terraform itself, rather than anything with this GitHub Action which installs Terraform. If you are still experiencing issues, please raise a fresh bug report and we can take a further look.
